src/openms/source/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.cpp

```diff
@@ -0,0 +1,456 @@
+      {
+        fs
+            << "FileName\tScanNum\tRetentionTime\tMassCountInSpec\tAverageMass\tMonoisotopicMass\t"
+               "SumIntensity\tMinCharge\tMaxCharge\t"
+               "PeakCount\tPeakMZs\tPeakIntensities\tPeakCharges\tPeakMasses\tPeakIsotopeIndices\tPeakPPMErrors\t"
+               "IsotopeCosine\tChargeScore\tMassSNR\tRepresentativeCharge\tMaxQScoreMzStart\tMaxQScoreMzEnd\tQScore\tPerChargeIntensity\tPerIsotopeIntensity\n";
+      }
+      else
+      {
+        fs
+            << "FileName\tScanNum\tRetentionTime\tMassCountInSpec\tAverageMass\tMonoisotopicMass\t"
+               "SumIntensity\tMinCharge\tMaxCharge\t"
+               "PeakCount\tPeakMZs\tPeakIntensities\tPeakCharges\tPeakMasses\tPeakIsotopeIndices\tPeakPPMErrors\t"
+               "PrecursorScanNum\tPrecursorMz\tPrecursorIntensity\tPrecursorCharge\tPrecursorMonoisotopicMass\tPrecursorQScore\t"
+               "IsotopeCosine\tChargeScore\tMassSNR\tRepresentativeCharge\tMaxQScoreMzStart\tMaxQScoreMzEnd\tQScore\tPerChargeIntensity\tPerIsotopeIntensity\n";
+      }
+    }
+    else
+    {
+      if (n == 1)
+      {
+        fs
+            << "FileName\tScanNum\tRetentionTime\tMassCountInSpec\tAverageMass\tMonoisotopicMass\t"
+               "SumIntensity\tMinCharge\tMaxCharge\t"
+               "PeakCount\t"
+               //"PeakMZs\tPeakCharges\tPeakMasses\tPeakIsotopeIndices\tPeakPPMErrors\tPeakIntensities\t"
+               "IsotopeCosine\tChargeScore\tMassSNR\tRepresentativeCharge\tMaxQScoreMzStart\tMaxQScoreMzEnd\tQScore\tPerChargeIntensity\tPerIsotopeIntensity\n";
+
+      }
+      else
+      {
+        fs
+            << "FileName\tScanNum\tRetentionTime\tMassCountInSpec\tAverageMass\tMonoisotopicMass\t"
+               "SumIntensity\tMinCharge\tMaxCharge\t"
+               "PeakCount\t"
+               "PrecursorScanNum\tPrecursorMz\tPrecursorIntensity\tPrecursorCharge\tPrecursorMonoisotopicMass\tPrecursorQScore\t"
+               "IsotopeCosine\tChargeScore\tMassSNR\tRepresentativeCharge\tMaxQScoreMzStart\tMaxQScoreMzEnd\tQScore\tPerChargeIntensity\tPerIsotopeIntensity\n";
+      }
+    }
+  }
+
+  /*  void DeconvolutedSpectrum::writeAttCsvHeader(std::fstream &fs)
+    {
+      fs
+          << "ScanNumber,RetentionTime,PrecursorScanNumber,Charge,ChargeSNR,PeakIntensity,EnvIntensity,EnvIsotopeCosine,PeakMz,"
+             "MonoMass,MassSNR,IsotopeCosine,MassIntensity,QScore,Class\n";
+    }*/
+
+  void DeconvolutedSpectrum::writeThermoInclusionHeader(std::fstream &fs)
+  {
+    fs << "Compound,Formula,Adduct,m/z,z,t start (min),t stop (min),Isolation Window (m/z),Normalized AGC Target (%)\n";
+  }
+
+
+  void DeconvolutedSpectrum::clearPeakGroupsChargeInfo()
+  {
+    for (auto &pg: *this)
+    {
+      pg.clearChargeInfo();
+    }
+  }
+
+  void DeconvolutedSpectrum::writeTopFD(std::fstream &fs, int id, const FLASHDeconvHelperStructs::PrecalculatedAveragine &avg)//, fstream &fsm, fstream &fsp)
+  {
+    auto msLevel = spec.getMSLevel();
```

Comment:
Please don't use auto for simple types like int or double

Reply:
That I did not know. I will apply this rule to all source codes..